src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -27,6 +38,91 @@ export interface RefetchQuoteCallbackParmams {
 
 type QuoteResult = [PromiseSettledResult<PriceInformation>, PromiseSettledResult<FeeInformation>]
 
+// const FEE_EXCEEDS_FROM_ERROR = new QuoteError({
+//   errorType: QuoteErrorCodes.FeeExceedsFrom,
+//   description: QuoteError.quoteErrorDetails.FeeExceedsFrom
+// })
+
+class PriceQuoteError extends Error {
+  params: PriceQuoteParams
+  results: PromiseSettledResult<any>[]
+
+  constructor(message: string, params: PriceQuoteParams, results: PromiseSettledResult<any>[]) {
+    super(message)
+    this.params = params
+    this.results = results
+  }
+}
+
+type PriceSource = 'gnosis-protocol' | 'paraswap'
+type PriceInformationWithSource = PriceInformation & { source: PriceSource; data?: any }
+type PromiseRejectedResultWithSource = PromiseRejectedResult & { source: PriceSource }
+
+/**
+ *
+ * @returns The best price among GPv2 API or Paraswap one
+ */
+async function _getBestPriceQuote(params: PriceQuoteParams): Promise<PriceInformation> {
+  // Get price from all API: Gpv2 and Paraswap
+  const pricePromise = withTimeout(getPriceQuote(params), PRICE_API_TIMEOUT_MS, 'GPv2: Get Price API')
+  const paraSwapPricePromise = withTimeout(
+    getPriceQuoteParaswap(params),
+    PRICE_API_TIMEOUT_MS,
+    'Paraswap: Get Price API'
+  )
+
+  // Get results from API queries
+  const [priceResult, paraSwapPriceResult] = await Promise.allSettled([pricePromise, paraSwapPricePromise])
+
+  // Prepare an array with all successful estimations
+  const priceQuotes: Array<PriceInformationWithSource> = []
+  const errorsGetPrice: Array<PromiseRejectedResultWithSource> = []
+
+  if (isPromiseFulfilled(priceResult)) {
+    priceQuotes.push({ ...priceResult.value, source: 'gnosis-protocol' })
```

Comment:
nit: the rename i suggested above makes this a breeze to read, ah it's the gnosis protocol price quotes being pushed, took me a seconds